pkg/jobs/runnable_job.go

```diff
@@ -61,14 +61,15 @@ func (r *runnableJob) Run(ctx context.Context) error {
 			if r.job.UID != newJob.UID {
 				return
 			}
-			if len(newJob.Status.Conditions) == 0 {
-				return
-			}
-			switch condition := newJob.Status.Conditions[0]; condition.Type {
-			case batchv1.JobComplete:
-				complete <- nil
-			case batchv1.JobFailed:
-				complete <- fmt.Errorf("job failed: %s: %s", condition.Reason, condition.Message)
+			for _, condition := range newJob.Status.Conditions {
+				switch condition.Type {
+				case batchv1.JobComplete, batchv1.JobSuccessCriteriaMet:
+					complete <- nil
+					return
+				case batchv1.JobFailed:
```

Comment:
Should we also check the `JobFailureTarget` condition?
From the [doc](https://kubernetes.io/docs/concepts/workloads/controllers/job/#termination-of-job-pods):
> You can use the FailureTarget or the SuccessCriteriaMet condition to evaluate whether the Job has failed or succeeded without having to wait for the controller to add a terminal condition.

Reply:
good catch! I thought about it, but decided to wait for `JobFailed`.  that means `all pods are terminated`.
to avoid an error https://github.com/aquasecurity/trivy/discussions/5639

wdyt?

Reply:
Thank you, that makes sense.

Reply:
thank you!